.github/workflows/hlint.yml

```diff
@@ -0,0 +1,24 @@
+name: hlint
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  build:
+    name: hlint
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up HLint
+      uses: haskell/actions/hlint-setup@v2
+      with:
+        version: "3.3.6"
+
+    - name: Run Hlint
+      uses: haskell/actions/hlint-run@v2
+      with:
+        path: '["src/", "test/"]'
+        fail-on: suggestion
```

Comment:
On reason for having a separate action is that hlint runs very quickly (doesn't require a build).

Reply:
This step could be done before the `Build` step

Reply:
I did this but in testing found it gave less feedback. I had to rely on trusting that running hlint locally would work the same as when run in the workflow. This can be seen by comparing [hlint](https://github.com/typechecker/colourista/actions/workflows/hlint.yml) and [hlint-standalone](https://github.com/typechecker/colourista/actions/workflows/hlint-standalone.yml) workflow runs. The `CI/hlint` job was last triggered when I edited the workflow whereas the `hlint-standalone/hlint-standalone` job was triggered when I made any change (hlint is lightweight so this doesn't cost much).